app/models/sponsorship.rb

```diff
@@ -41,4 +31,20 @@ class Sponsorship < ApplicationRecord
 
   scope :live, -> { where(status: :live) }
   scope :pending, -> { where(status: :pending) }
+
+  private
+
+  def validate_tag_uniqueness
+    return unless self.class.where(sponsorable: sponsorable, level: :tag).
+      where("expires_at > ? AND id != ?", Time.current, id.to_i).exists?
+
+    errors.add(:level, "The tag is already sponsored")
+  end
+
+  def validate_level_uniqueness
+    return unless self.class.where(organization: organization).
+      where("level IN (?) AND expires_at > ? AND id != ?", LEVELS_WITH_EXPIRATION, Time.current, id.to_i).exists?
```

Comment:
`LEVELS_WITH_EXPIRATION` name became misleading because currently, tag sponsorships have `expires_at` which is set on creation as well. I decided not to touch this issue in this pr and leave it for the future, we'll need more refactoring/improvements anyway.